apollo-router/src/plugins/telemetry/config_new/graphql/mod.rs

```diff
@@ -222,9 +228,28 @@ struct GraphQLInstrumentsVisitor<'a> {
     instruments: &'a GraphQLInstruments,
 }
 
-impl<'a> Visitor for GraphQLInstrumentsVisitor<'a> {
-    fn visit_field(&self, value: &TypedValue) {
-        self.instruments.on_response_field(value, self.ctx);
+impl<'a> ResponseVisitor for GraphQLInstrumentsVisitor<'a> {
+    fn visit_field(
+        &mut self,
+        request: &ExecutableDocument,
+        ty: &NamedType,
+        field: &Field,
+        value: &Value,
+    ) {
+        self.instruments
```

Comment:
This is required if we want to capture list length, but it also means `field` gets overcounted when it's also passed into the list items below. We could account for this in the attributes, but I don't love the idea of this implementation detail leaking into the metrics code.